great_expectations/experimental/datasource/fabric.py

```diff
@@ -0,0 +1,394 @@
+from __future__ import annotations
+
+import os
+import uuid
+from pprint import pformat as pf
+from typing import (
+    TYPE_CHECKING,
+    ClassVar,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Set,
+    Tuple,
+    Type,
+    Union,
+)
+
+from typing_extensions import Annotated, TypeAlias
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations.compatibility import pydantic
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.core.batch_spec import FabricBatchSpec
+from great_expectations.datasource.fluent import BatchRequest
+from great_expectations.datasource.fluent.constants import _DATA_CONNECTOR_NAME
+from great_expectations.datasource.fluent.interfaces import (
+    Batch,
+    DataAsset,
+    Datasource,
+    Sorter,
+    TestConnectionError,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch_spec import FabricReaderMethods
+    from great_expectations.datasource.fluent.interfaces import (
+        BatchMetadata,
+    )
+    from great_expectations.execution_engine import PandasExecutionEngine
+
+SortersDefinition: TypeAlias = List[Union[Sorter, str, dict]]
+
+
+class _PowerBIAsset(DataAsset):
+    """Microsoft PowerBI Asset base class."""
+
+    _reader_method: ClassVar[FabricReaderMethods]
+    _EXCLUDE_FROM_READER_OPTIONS: ClassVar[Set[str]] = {
+        "batch_metadata",
+        "name",
+        "order_by",
+        "type",
+        "id",
+    }
+
+    @override
+    def test_connection(self) -> None:
+        """
+        Whatever is needed to test the connection to and/or validatitly of the asset.
+        This could be a noop.
+        """
+        pass
```

Comment:
Do we want a complete noop or should we log something?

Reply:
I'll add a debug level log.